xl-v1.go

```diff
@@ -429,8 +429,26 @@ func (xl XL) StatFile(volume, path string) (FileInfo, error) {
 		return FileInfo{}, errInvalidArgument
 	}
 
+	readLock := true
```

Comment:
> Yes, I understand now.. but won't that lead to unexpected behaviours somewhere else? in that case, you should make sure that routines which are waiting for the lock exit too with preserving the system state consistent

Yes it would, it requires a more granular locking.  The statements like "lock" and "defer unlock" cause unnecessary regions to be locked, it requires testing and figuring out right locations. We will see if channels can be used as well.